### ISSUE TITLE
Even smaller script causes stack owerflow

### ISSUE THREAD

**Issue by abenkovskii** [KSP-KOS/KOS#885]
Q: Why Even smaller?
A: It's a reference to: #879
Q: What versions are affected?
A:
0.17.0 - works fine
latest develop - fails

Code:
```
// file bug.ks
function foo
{
parameter x.
}
```
```
// enter it into the terminal:
run bug.
foo(42).
```
Sounds familiar but I'm 99% sure it's different: #858
output_log: https://www.dropbox.com/s/n90afkcbu5az1cc/output_log_stack_owerflow_2.txt?dl=0

**Comment by Dunbaratu** [comment]
Given the opcodes in the output log I'm suspicious that you aren't showing the whole story here.
It looks as if you attempted to call foo() once before running bug.  I'm going to give this a test but in any case the use case isn't supported anyway (calling functions in a script from the interpreter).  But it should be doing something better than a stack overflow when it happens.

**Comment by Dunbaratu** [comment]
I just tried the exact steps as shown, and got a totally different result.
I got, as expected:
"Undefined Variable name "$foo*".
BUT, also, weirdly, It popped open the editor again.  It re-ran the EDIT BUG command, which is definitely wrong.

**Comment by Dunbaratu** [comment]
Yup, when I inserted the extra command you didn't mention but was evidenced in the output_log, THEN I got the result you showed:
```
edit bug.
foo(). // the missing statement from the description.  It produces an Undefined Variable Name.
run bug.
foo(42). // This produces stack error now.
```

In both cases the problem is essentially the already known one mentioned in issue #717.

You cannot call functions declared in programs from inside the interpreter.  There should be a better way to give the error message though.  What's happening now is that the global variable ``$foo*`` still contains the delegate referring to the memory location where the function's code used to exist inside the PROGRAM's context instead of inside the INTERPRETER's context.  So if the PROGRAM had the function at memory location, say, 7, then it tries to call whatever randomly happens to be in the INTERPRETER's context at location 7, which will be something totally different and dependent on what you've run earlier.

I may add a record of which context (program or interpreter) the delegate was created from to the delegate object.  Then it could at least detect when you're trying to call it from the wrong context and give a better error message.

Then later that same thing can be expanded to actually try to implement fixing it to make it work.

**Comment by abenkovskii** [comment]
0) I know that calling functions from files is not yet implemented but the error changed since release.

**Comment by abenkovskii** [comment]
1) Similar situation but with my lib_exec:
the result is different: constantly spams an error into terminal, ctrl + C
toggling power fixes it.
![screenshot0](https://cloud.githubusercontent.com/assets/11031560/7239557/8cd67f46-e7b3-11e4-97c1-23f9bc06555b.png)
log: https://www.dropbox.com/s/j4zhlikufbj823r/output_log_1234567890.txt?dl=0

**Comment by Dunbaratu** [comment]
I have no evidence that the error changed since release.  See above.  This is exactly the behavior being mentioned in issue #717.

The actual error message you'll see will vary depending on what else is in your interpreter context, where the functions existed in the program context, whether your interpreter context has grown large enough to even have a memory location that large yet to jump to or not (if it hasn't then you get an array index out of range trying to jump to instruction N when the memory doesn't go up that high).

It's one of those cases where because the real problem is a jump to totally the wrong place in the wrong program even, there is no consistency in the kinds of problems you end up seeing.

Again, this is NOT new.  It's a problem, but just because you see a different error message than before does not make it NEW.  The problem already known about when we released 0.17.0, called issu #717, *already* is exactly the one you are encountering.

**Comment by Dunbaratu** [comment]
Closed as a duplicate of #717.  Further discussion about what to do about the problem belongs under #717.
